packages/astro/src/core/errors/errors-data.ts

```diff
@@ -693,6 +693,35 @@ See https://docs.astro.build/en/guides/server-side-rendering/ for more informati
 			'`locals` can only be assigned to an object. Other values like numbers, strings, etc. are not accepted.',
 		hint: 'If you tried to remove some information from the `locals` object, try to use `delete` or set the property to `undefined`.',
 	},
+
+	/**
+	 * @docs
+	 * @see
+	 * - [Astro.glob](https://docs.astro.build/en/reference/api-reference/#astroglob)
+	 * @description
+	 * `Astro.glob()` does not work outside of an Astro file. You can use [`import.meta.glob()`](https://vitejs.dev/guide/features.html#glob-import) instead to acheive the same result.
+	 */
+	AstroGlobUsedOutside: {
+		title: 'Astro.glob() used outside of an Astro file',
+		code: 3034,
+		message: (globStr: string) =>
+			`\`Astro.glob(${globStr})\` does not work outside of an Astro file. You can use \`import.meta.glob(${globStr})\` instead to acheive the same result.`,
```

Comment:
The message is a bit misleading, because `Astro.glob` isn't 100% equal to `import.meta.glob`, since they don't return the same shape

```suggestion
			`\`Astro.glob(${globStr})\` cannot be used outside Astro files. \`import.meta.glob(${globStr})\` can be used instead to achieve a similar result.`,
			hint: "See Vite's documentation on `import.meta.glob` for more information: https://vitejs.dev/guide/features.html#glob-import",
```